Source/Core/DolphinQt/GameList/GameTracker.cpp

```diff
@@ -341,8 +312,8 @@ void GameTracker::LoadGame(const QString& path)
   }
 }
 
-void GameTracker::PurgeCache()
+void GameTracker::OnPurgeCache()
```

Comment:
If the list is currently updating the "Purge Game List Cache" action seems to just be queued and shows no response.
This isn't good UX.
Purging the cache should abort the current refresh operation and purge the cache immediately.

Reply:
I totally oversaw this one. It is indeed a problem; very obvious if a sleep is added in `GameTracker::LoadGame()`. This issue predates this PR as well.

Reply:
As mentioned in the other comment, in order to halt the loop in `UpdateDirectory()`, I've added the `m_pendingTask` flag, that is set when purging or forcing a refresh.